Minio/Credentials/IAMAWSProvider.cs

```diff
@@ -50,9 +44,9 @@ public IAMAWSProvider(string endpoint, MinioClient client)
                 throw new ArgumentNullException("Endpoint field " + nameof(CustomEndPoint) + " is invalid.");
         }
 
-        if (client == null)
-            throw new ArgumentException("MinioClient reference field " + nameof(Minio_Client) + " cannot be null.");
-        Minio_Client = client;
+        Minio_Client = client ??
+                       throw new ArgumentException("MinioClient reference field " + nameof(Minio_Client) +
```

Comment:
```suggestion
                       throw new ArgumentException("MinioClient reference field " +
                            nameof(Minio_Client) + " cannot be null.");
```

Maybe the firist line shorter is better.